lib/si-frontend-mv-types-rs/src/materialized_view.rs

```diff
@@ -1,9 +1,67 @@
-use si_events::workspace_snapshot::EntityKind;
+use si_events::workspace_snapshot::{
+    Checksum,
+    ChecksumHasher,
+    EntityKind,
+};
 
 use crate::reference::ReferenceKind;
 
 pub trait MaterializedView {
     fn kind() -> ReferenceKind;
     fn reference_dependencies() -> &'static [ReferenceKind];
     fn trigger_entity() -> EntityKind;
+    fn definition_checksum() -> Checksum;
 }
+
+#[derive(Debug, Clone)]
+pub struct MaterializedViewInventoryItem {
+    kind: ReferenceKind,
+    reference_dependencies: &'static [ReferenceKind],
+    trigger_entity: EntityKind,
+    definition_checksum: &'static ::std::sync::LazyLock<Checksum>,
+}
+
+impl MaterializedViewInventoryItem {
+    pub const fn new(
+        kind: ReferenceKind,
+        reference_dependencies: &'static [ReferenceKind],
+        trigger_entity: EntityKind,
+        definition_checksum: &'static ::std::sync::LazyLock<Checksum>,
+    ) -> Self {
+        MaterializedViewInventoryItem {
+            kind,
+            reference_dependencies,
+            trigger_entity,
+            definition_checksum,
+        }
+    }
+
+    pub fn kind(&self) -> ReferenceKind {
+        self.kind
+    }
+
+    pub fn reference_dependencies(&self) -> &'static [ReferenceKind] {
+        self.reference_dependencies
+    }
+
+    pub fn trigger_entity(&self) -> EntityKind {
+        self.trigger_entity
+    }
+    pub fn definition_checksum(&self) -> Checksum {
+        **self.definition_checksum
+    }
+}
+
+pub fn materialized_view_definitions_checksum() -> Checksum {
+    let mut mv_items: Vec<_> = ::inventory::iter::<MaterializedViewInventoryItem>().collect();
+    mv_items.sort_by_key(|item| item.kind());
+
+    let mut hasher = ChecksumHasher::new();
+    for mv in mv_items {
+        hasher.update(mv.definition_checksum().as_bytes());
+    }
+
+    hasher.finalize()
+}
```

Comment:
This could probably be memoized via `LazyStatic`, but doing the checksum calculation here should be really fast. Thinking about making that change when addressing the `ActionViewList`/`ActionView` limitation, if it's not handled by then, but I could probably be talked into doing the `LazyStatic` change in this PR without much convincing.